### PR TITLE
Refactor certificate and user GET/DELETE methods. Add DELETE api/certifacte routes

### PR DESCRIPTION
**Pull request by debus**
Relies on https://github.com/ImageWare/TLSential/pull/40 due to changes in api/api.go
Addresses https://github.com/ImageWare/TLSential/issues/38
## This PR addresses the following issues: 

adding `DELETE /api/certificate` and `DELETE /api/certificate/{id}` routes
Splitting up certificate.Delete into Delete and DeleteAll
Same for user.Delete and user.Get

### Testing

Hit the URLs and made sure the responses were sane.
`DELETE /api/certificate` => `No Content`
`DELETE /api/certificate/someid` => `404`
`GET /api/user` => `[{my_user}]`
`GET /api/user/my_user` => `{my_user}`
`GET /api/user/my_user_but_not` => `404`
`DELETE /api/user/my_user_but_not` => `404`
`DELETE /api/user` => `No Content`

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -122,33 +124,33 @@ func (h *userHandler) Delete() http.HandlerFunc {
...
+
+		var urs = make([]*UserResp, 0)
+		for _, u := range users {
+			ur := &UserResp{Name: u.Name, Role: u.Role}
+			urs = append(urs, ur)
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(urs)
```
Comment (d1str0) [inline api/user.go]: Can you go through and make sure all of these calls check for err?
https://github.com/ImageWare/TLSential/blob/develop/api/certificate.go#L150

**Review by d1str0 — CHANGES_REQUESTED** [review]
There are several places where json.NewEncoder(w).Encode(value) aren't checked for errors. This line could potentially return an error and if it does it is most likely a server error. Please log and return 500.

Example: https://github.com/ImageWare/TLSential/blob/develop/api/certificate.go#L150

File: api/user.go (hunk 1 above)
Comment (debus) [reply api/user.go]: Pushed an update. I believe I got them all

**Comment by d1str0** [comment]
Lgtm

**Review by d1str0 — APPROVED** [review]
🤝

**Pull request merged**